src/components/hosts/AdditionalNTPSourcesDialog.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import {
+  Alert,
+  Form,
+  Modal,
+  ModalVariant,
+  ModalBoxBody,
+  ModalBoxFooter,
+  Button,
+  AlertActionCloseButton,
+  AlertVariant,
+  AlertActionLink,
+} from '@patternfly/react-core';
+import { Cluster, ClusterUpdateParams } from '../../api/types';
+import { ntpSourceValidationSchema } from '../ui/formik/validationSchemas';
+import TextAreaField from '../ui/formik/TextAreaField';
+import { patchCluster } from '../../api/clusters';
+import { updateCluster } from '../../features/clusters/currentClusterSlice';
+import { getErrorMessage, handleApiError } from '../../api/utils';
+import { trimCommaSeparatedList } from '../ui/formik/utils';
+
+type AdditionalNTPSourcesDialogToggleProps = {
+  cluster: Cluster;
+  onToggle?: (isOpen: boolean) => void;
```

Comment:
If i look correctly, `onToggle` can be set as a mandatory field.

Reply:
explained below